runtime/bcverify/vrfyhelp.c

```diff
@@ -931,10 +940,17 @@ isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *f
 				 * the field is accessible only when it exists in the subclass or the superclass's
 				 * initialization is completed in which case uninitializedThis is set to FALSE.
 				 */
-				J9BranchTargetStack *liveStack = (J9BranchTargetStack *)verifyData->liveStack;
-				return findFieldFromCurrentRomClass(romClass, fieldRef) || !liveStack->uninitializedThis;
+				return (NULL != field) || !liveStack->uninitializedThis;
 			}
 		}
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+		else {
+			/* putfield is not allowed outside of initialization for strict fields. */
+			if ((NULL != field) && J9_ARE_ALL_BITS_SET(field->modifiers, J9AccStrict)) {
```

Comment:
How about simplify it as follows:
```
else if ((NULL != field) && J9_ARE_ALL_BITS_SET(field->modifiers, J9AccStrict)) {
...
}
```

Reply:
Fixed.

Reply:
Really appreciate your quick response. 

Maybe it's better to move it to the front to avoid duplicate the statement. e.g.
```
isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *fieldRef, UDATA bytecode, UDATA receiver, IDATA *reasonCode)
{
...
IDATA isStrictField = (NULL != findFieldFromCurrentRomClass(romClass, fieldRef)) && J9_ARE_ALL_BITS_SET(field->modifiers, J9AccStrict); 
....
	if (J9_ARE_ALL_BITS_SET(receiver, BCV_SPECIAL_INIT)) {
#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
                        /* ACC_STRICT field must be assigned before instance initialization method. */
			if (isStrictField && !liveStack->uninitializedThis) {
                                  return (IDATA)FALSE;
			}
...
#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
		else if (isStrictField) { /* putfield is not allowed outside of initialization for strict fields. */
			return (IDATA)FALSE;
		}
#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
```